On armv7-unknown-linux-gnueabihf systems like RPi 3, The values needed are u32.
However for Alpine 32bit systems (using musl), u64 is needed.

diff --git a/src/builtins/ulimit.rs b/src/builtins/ulimit.rs
index d528e10..d72510e 100644
--- a/src/builtins/ulimit.rs
+++ b/src/builtins/ulimit.rs
@@ -79,15 +79,9 @@ fn set_limit(limit_name: &str, value: u64, for_hard: bool) -> String {
 
     // to support armv7-unknown-linux-gnueabihf
     if for_hard {
-        #[cfg(target_pointer_width = "32")]
-        { rlp.rlim_max = value as u32; }
-        #[cfg(target_pointer_width = "64")]
-        { rlp.rlim_max = value; }
+        rlp.rlim_max = value;
     } else {
-        #[cfg(target_pointer_width = "32")]
-        { rlp.rlim_cur = value as u32; }
-        #[cfg(target_pointer_width = "64")]
-        { rlp.rlim_cur = value; }
+        rlp.rlim_cur = value;
     }
 
     unsafe {
